pipewire/qubes-pw-module.c

```diff
@@ -1691,6 +1767,27 @@ int pipewire__module_init(struct pw_impl_module *module, const char *args)
     if ((res = pw_impl_module_update_properties(module, &SPA_DICT_INIT_ARRAY(module_props))) < 0)
         goto error;
 
+    {
+        int domid_from_props = get_domid_from_props(props);
+        // If domid is not specified from pipewire module context,
+        // then start the qubesdb watcher
+        if (domid_from_props<0) {
+            qdb_handle_t qdb = qdb_open(NULL);
+            if (!qdb) {
+                res = -errno;
+                pw_log_error("Could not open QubesDB");
+                goto error;
+            }
+            impl->qdb = qdb;
+            impl->domid = get_domid_from_qdb(impl->qdb);
```

Comment:
Getting domid should be done just after setting up the watch, otherwise you'll miss modification between those two calls.